processing/src/main/java/org/apache/druid/segment/filter/BoundFilter.java

```diff
@@ -166,6 +164,12 @@ public ColumnIndexCapabilities getIndexCapabilities()
         return rangeIndex.getIndexCapabilities().merge(nullBitmap.getIndexCapabilities());
       }
 
+      @Override
+      public int estimatedComputeCost()
+      {
+        return rangeIndex.estimatedComputeCost();
```

Comment:
nit: this probably should be 
```suggestion
        return rangeIndex.estimatedComputeCost() + 1;
```
since we are adding the extra union operation to add the null value bitmap